pkg/ebpf/telemetry/stack_sizes.go

```diff
@@ -0,0 +1,129 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package telemetry
+
+import (
+	"debug/elf"
+	"encoding/binary"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/native"
+	"github.com/cilium/ebpf"
+)
+
+const (
+	lowerOrder7Mask = 0x7f
+	higherOrderBit  = 0x80
+	bpfStackLimit   = 512
+)
+
+// https://en.wikipedia.org/wiki/LEB128
+func decodeULEB128(in io.Reader) (uint64, error) {
+	var result, shift uint64
+	for {
+		b := []byte{0}
+		if _, err := in.Read(b); err != nil {
```

Comment:
You need to handle the `n` return value first, before processing `err`. You also need to check for EOF. See https://pkg.go.dev/io@go1.22.1#Reader